Jenkinsfile

```diff
@@ -0,0 +1,203 @@
+#!/usr/bin/env groovy
+// vim: set et sw=4 ts=4 :
+pipeline {
+    agent any
+    options {
+        disableConcurrentBuilds() // Otherwise clean would delete the images
+        skipDefaultCheckout() // We do our own checkout so it can be disabled
+        timestamps()
+        timeout(time: 5, unit: 'HOURS')
+    }
+    parameters {
+        booleanParam(
+            name: 'SKIP_CHECKOUT',
+            defaultValue: false,
+            description: 'Skip the checkout step for faster iteration',
+        )
+        booleanParam(
+            name: 'RESET',
+            defaultValue: false,
+            description: 'Remove all existing build artifacts',
+        )
+        booleanParam(
+            name: 'CLEAN',
+            defaultValue: true,
+            description: 'Remove build artifacts that should normally not be reused',
+        )
+        booleanParam(
+            name: 'PUBLISH',
+            defaultValue: true,
+            description: 'Enable publishing',
+        )
+        credentials(
+            name: 'S3_CREDENTIALS',
+            description: 'AWS access key / secret key used for publishing',
+            defaultValue: 'cred-s3-scf',
+        )
+        string(
+            name: 'S3_REGION',
+            description: 'AWS S3 region the target bucket is in',
+            defaultValue: 'us-east-1',
+        )
+        string(
+            name: 'S3_BUCKET',
+            description: 'AWS S3 bucket to publish to',
+            defaultValue: 'scf',
+        )
+        string(
+            name: 'S3_PREFIX',
+            description: 'AWS S3 prefix to publish to',
+            defaultValue: 'scf/config/',
+        )
+        credentials(
+            name: 'DOCKER_CREDENTIALS',
+            description: 'Docker credentials used for publishing',
+            defaultValue: 'cred-docker-scf',
```

Comment:
This is the id of the credential. It has a default value, otherwise when Jenkins makes a build (on a new commit / PR / whatever), it tries to start and falls over due to not having a value.  Yes, it's that dumb.